### PR TITLE
fix: use and mock lodash debounce in Carousel and Dropdown

### PR DESCRIPTION
**Pull request by silviuaavram**
Removed in-house implementation of `debounce` and replaced its uses with `_.debounce` in `Dropdown` and `Carousel`.

Properly mocked the function in `Dropdown-test` as it wasn't working with jest timers. Details in https://github.com/facebook/jest/issues/3465 and other similar Github issues.

### CODE REVIEW

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -361,15 +361,8 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
...
   }
 
-  a11yStatusTimeout: any
-  charKeysPressedTimeout: any
   defaultTriggerButtonId = _.uniqueId('dropdown-trigger-button-')
 
-  componentWillUnmount() {
-    clearTimeout(this.a11yStatusTimeout)
-    clearTimeout(this.charKeysPressedTimeout)
-  }
```
Comment (layershifter) [inline packages/react/src/components/Dropdown/Dropdown.tsx:371]: Please add an `cancel()` call for `_.debounce`:

https://lodash.com/docs/4.17.15#debounce

Comment (silviuaavram) [reply packages/react/src/components/Dropdown/Dropdown.tsx:371]: you mean `this.clearStartingString.cancel()` ?

File: packages/react/src/components/Carousel/Carousel.tsx [hunk 2]
```diff
@@ -213,8 +212,7 @@ class Carousel extends AutoControlledComponent<WithAsProp<CarouselProps>, Carous
   itemRefs = [] as React.RefObject<HTMLElement>[]
   paddleNextRef = React.createRef<HTMLElement>()
   paddlePreviousRef = React.createRef<HTMLElement>()
-
-  focusItemAtIndex = debounce((index: number) => {
+  focusItemAtIndex = _.debounce((index: number) => {
     this.itemRefs[index].current.focus()
```
Comment (dzearing) [inline packages/react/src/components/Carousel/Carousel.tsx]: If you're debouncing, you'll need to check `current` for null. There is a slight race condition where the debounced callback can occur asynchronously after unmounting.

File: packages/react/src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (dzearing) [reply packages/react/src/components/Dropdown/Dropdown.tsx:371]: I believe `this.focusItemAtIndex.cancel()` as well

File: packages/react/src/components/Carousel/Carousel.tsx [hunk 3]
```diff
@@ -213,8 +212,7 @@ class Carousel extends AutoControlledComponent<WithAsProp<CarouselProps>, Carous
   itemRefs = [] as React.RefObject<HTMLElement>[]
   paddleNextRef = React.createRef<HTMLElement>()
   paddlePreviousRef = React.createRef<HTMLElement>()
-
-  focusItemAtIndex = debounce((index: number) => {
+  focusItemAtIndex = _.debounce((index: number) => {
```
Comment (dzearing) [inline packages/react/src/components/Carousel/Carousel.tsx:219]: Minor nullref edge case if you don't cancel this.

File: packages/react/src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (silviuaavram) [reply packages/react/src/components/Dropdown/Dropdown.tsx:371]: Solved. Thanks!

**Review by layershifter — APPROVED** [review]

**Pull request merged**